### PR TITLE
Dackbox imagecomponent and nodecomponent read tests

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

Adding tests for scope checks on image components and node components.
The goal is to ensure the datastore behaviour is the same before and after the migration to postgres.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Adding unit tests for dackbox imagecomponent and nodecomponent

### CODE REVIEW

File: central/imagecomponent/datastoretest/nodeomponent_from_imagecomponent_datastore.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package datastoretest
```
[the remark this answers is not in the document]
Comment (connorgorman) [reply central/imagecomponent/datastoretest/nodeomponent_from_imagecomponent_datastore.go:1]: Also, I don't understand the purpose of this file

File: central/role/resources/list.go [hunk 2]
```diff
@@ -38,8 +38,6 @@ var (
 	DeploymentExtension = newResourceMetadata("DeploymentExtension", permissions.NamespaceScope)
 	Detection           = newResourceMetadata("Detection", permissions.GlobalScope)
 	Image               = newResourceMetadata("Image", permissions.NamespaceScope)
-	// SAC check is not performed directly on ImageComponent resource. It exists here for postgres sac generation to pass.
-	ImageComponent = newResourceMetadata("ImageComponent", permissions.NamespaceScope)
```
Comment (md2119) [inline central/role/resources/list.go:42]: ✅

File: central/imagecomponent/datastoretest/nodeomponent_from_imagecomponent_datastore.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhybrillou) [reply central/imagecomponent/datastoretest/nodeomponent_from_imagecomponent_datastore.go:1]: Nice catch, thanks.

File: central/imagecomponent/datastoretest/nodecomponent_from_imagecomponent_datastore.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	searchPkg "github.com/stackrox/rox/pkg/search"
+)
+
+type nodeComponentFromImageComponentDataStore struct {
+	imageComponentStore imageComponentDataStore.DataStore
+}
+
+func convertImageComponentToNodeComponent(imageComponent *storage.ImageComponent) *storage.NodeComponent {
+	return &storage.NodeComponent{
+		Id:              imageComponent.Id,
```
Comment (md2119) [inline central/imagecomponent/datastoretest/nodecomponent_from_imagecomponent_datastore.go:18]: Just an FYI that the ID of components in rocksdb and in postgres is different.

File: central/imagecomponent/datastoretest/datastore_sac_test.go [hunk 4]
```diff
@@ -0,0 +1,795 @@
...
+	s.Require().NoError(s.dackboxTestStore.CleanNodeToVulnerabilitiesGraph(waitForIndexing))
+}
+
+type componentTestCase struct {
+	contextKey             string
+	expectedComponentFound map[string]bool
+}
+
+var (
+	image1OS              = fixtures.GetImageSherlockHolmes1().GetScan().GetOperatingSystem()
```
Comment (md2119) [inline central/imagecomponent/datastoretest/datastore_sac_test.go:87]: :)

**Review by md2119 — APPROVED** [review]

**Pull request merged**